app/controllers/api/v0/manifests_controller.rb

```diff
@@ -12,30 +8,31 @@ def search
   end
 
   def validate
-    begin
-      manifest_content = JSON.parse(request.body.read)
+    manifest_content = read_body_as_json
+    unless performed?
       if validate_manifest(manifest_content)
         render json: {message: "Manifest structure is valid"}, status: 200
       end
-    rescue JSON::ParserError => error
-      render json: {message: "Invalid JSON in request: #{error}"}, status: 400
     end
   end
 
   def create
-    if validate_manifest(manifest_params)
-      @manifest = Manifest.new(content: manifest_params)
+    manifest_content = read_body_as_json
+    unless performed?
+      if validate_manifest(manifest_content)
+        @manifest = Manifest.new(content: manifest_content)
 
-      if @manifest.save
-        tracking_number = manifest_params[:manifest_tracking_number]
-        render json: {
-          message: "Manifest #{tracking_number} submitted successfully.",
-        }, status: 201
-      else
-        render json: {
-          message: "Validation failed",
-          errors: @manifest.errors.full_messages.to_sentence
-        }, status: 422
+        if @manifest.save
```

Comment:
Right now, you are correct. Since the only validation in the model is redundant with what is done via the JSON schema, we should not see the case you're describing.

However, I don't think it's a bad idea for more error checking. The alternative would be calling `@manifest.save!` in which case something like a psql error should also surface (not just validation errors but anything that fails in the INSERT or UPDATE).